### DISCUSSION TITLE
Appropriate way to handle dynamic rules with Controller

### DISCUSSION THREAD

**Discussion by ryanjohndunne** [react-hook-form/react-hook-form#4889] [Q&A]
I'm trying to change the required rule to false/true depending on a switch which hides input fields.

You can see the example here: https://codesandbox.io/s/condescending-currying-sqvx1?file=/src/App.js:3434-3444

Steps:
1. Put some value into test field 1, leaving test field 2 empty
2. Hit the switch
3. Put some value into test field 3 and test field 4
4. press submit button

Notice that you will not get anything logged to the console because required is set to true on test field 2 even though I've attempted to unregister it. Is there an appropriate way to handle this use case?

Any help would be appreciated, thank you.

**Comment by bluebill1049** [comment]
i think write some logic inside `validate` function would be much simpler.

**Reply by ryanjohndunne** [reply]
Is there any examples of validate being used with Controller? I can only find them with register.

**Reply by bluebill1049** [reply]
```
rules={{
  validate: (value) => {
    if (a && value === 'test') // do something
    // or do else
  }
}}
```

this helps?
